airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/request_headers/interpolated_request_header_provider.py

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+from typing import Any, Mapping
+
+from airbyte_cdk.sources.declarative.requesters.interpolated_request_input_provider import InterpolatedRequestInputProvider
+from airbyte_cdk.sources.declarative.requesters.request_headers.request_header_provider import RequestHeaderProvider
+
+
+class InterpolatedRequestHeaderProvider(RequestHeaderProvider):
+    """
+    Provider that takes in a dictionary of request headers and performs string interpolation on the defined templates and static
+    values based on the current state of stream being processed
+    """
+
+    def __init__(self, *, config, request_headers):
+        self._interpolator = InterpolatedRequestInputProvider(config=config, request_inputs=request_headers)
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return self._interpolator.request_inputs(stream_state, stream_slice, next_page_token)
```

Comment:
This code became literally identical hence why I moved it all into one place that both header and provider can share.